src/battle/battle_command.c

```diff
@@ -5909,3 +5916,561 @@ static void *BattleScriptGetVarPointer(BattleSystem *bsys, BattleContext *ctx, i
+    } else if ((battleType & BATTLE_TYPE_DOUBLES)
+            && (battleType & BATTLE_TYPE_6) == FALSE
+            && data->ctx->selectedMonIndex[BATTLER_PLAYER2] == slot) {
+        expBattler = 2;
+    }
+
+    switch (data->state) {
+    case STATE_GET_EXP_START: {
+        item = GetMonData(mon, MON_DATA_HELD_ITEM, NULL);
+        itemEffect = GetItemAttr(item, ITEM_VAR_HOLD_EFFECT, HEAP_ID_BATTLE);
+
+        // Declare victory if all wild mons have been defeated
+        if ((battleType & BATTLE_TYPE_TRAINER) == FALSE
+                && data->ctx->battleMons[BATTLER_ENEMY].hp
+                    + data->ctx->battleMons[BATTLER_ENEMY2].hp == 0
+                && GetMonData(mon, MON_DATA_HP, NULL)
+                && data->ctx->unk_3144 == FALSE) {
+            PlayBGM(SEQ_GS_WIN2);
+            data->ctx->unk_3144 = TRUE;
+            BattleSystem_SetCriticalHpMusicFlag(data->bsys, 2); // turn off
+        }
+
+        u32 totalExp = 0;
+        msg.id = 1; // "{0} gained {1} Exp. Points!"
+
+        if (GetMonData(mon, MON_DATA_HP, NULL) && GetMonData(mon, MON_DATA_LEVEL, NULL) != 100) {
+            if (data->ctx->unk_A4[battler] & MaskOfFlagNo(slot)) {
+                totalExp = data->ctx->gainedExp;
+            }
+
+            if (itemEffect == HOLD_EFFECT_EXP_SHARE) {
+                totalExp += data->ctx->partyGainedExp;
+            }
+
+            if (itemEffect == HOLD_EFFECT_EXP_UP) {
+                totalExp = totalExp * 150 / 100;
+            }
+
+            if (battleType & BATTLE_TYPE_TRAINER) {
+                totalExp = totalExp * 150 / 100;
+            }
+
+            if (ov12_022568B0(data->bsys, mon) == FALSE) {
+                if (GetMonData(mon, MON_DATA_GAME_LANGUAGE, NULL) != gGameLanguage) {
+                    totalExp = totalExp * 170 / 100;
+                } else {
+                    totalExp = totalExp * 150 / 100;
+                }
+
+                msg.id = 2; // "{0} gained a boosted {1} Exp. Points!"
+            }
+
+            u32 newExp = GetMonData(mon, MON_DATA_EXPERIENCE, NULL);
+            data->unk30[3] = newExp - GetMonBaseExperienceAtCurrentLevel(mon);
+            newExp += totalExp;
+
+            if (slot == data->ctx->selectedMonIndex[expBattler]) {
+                data->ctx->battleMons[expBattler].exp = newExp;
+            }
+
+            SetMonData(mon, MON_DATA_EXPERIENCE, &newExp);
+            ov12_022463E8(BattleSystem_GetParty(data->bsys, expBattler),
+                    slot,
+                    data->ctx->battleMons[data->ctx->battlerIdFainted].species,
+                    data->ctx->battleMons[data->ctx->battlerIdFainted].form);
+        }
+
+        if (totalExp) {
+            msg.tag = 17;
+            msg.param[0] = expBattler | (slot << 8);
+            msg.param[1] = totalExp;
+            data->unk30[0] = ov12_0223C494(data->bsys, msgLoader, &msg, BattleSystem_GetTextFrameDelay(data->bsys));
+            data->unk30[1] = 30 / 4;
```

Comment:
Is 30 or 4 significant? Why not simply `data->unk30[1] = 7;`?

Reply:
Pretty sure 30 is a standard delay factor after a message. In this case, 30 / 4 would signify that it waits 7 frames to continue after the message has finished printing.

I didn't provide a constant for that because I wasn't 100% sure on that when I wrote the platinum code, which this PR is a port of.